### PR TITLE
Spark 3.0 / Scala 2.12 compatible build

### PR DESCRIPTION
**Pull request by mengxr**
Make a Spark 3.0 / Scala 2.12 compatible build based on #575 from @WeichenXu123 . The build and tests should pass but with the following issues:

* xgboost projects are skipped due to no Scala 2.12 jars
* no model backward compatibility on models with behavior change
* no cross spark 2.x/3.0 build (tests won't pass against Spark 2.x)

We need to branch based on Spark version to ensure backward compatibility, which would require more work.

Closes #575

### CODE REVIEW

File: mleap-core/src/test/scala/ml/combust/mleap/core/feature/MaxAbsScalerModelSpec.scala [hunk 1]
```diff
@@ -13,8 +13,7 @@ class MaxAbsScalerModelSpec extends FunSpec {
 
     it("Scales the vector based on absolute max value"){
       val inputArray = Array(15.0, -5.0, 5.0, 19.0)
-      val expectedVector = Array(0.75, -0.5, 0.5, 0.95)
-
+      val expectedVector = Array(0.75, -0.5, 0.5, 0.9500000000000001)
```
Comment (WeichenXu123) [inline mleap-core/src/test/scala/ml/combust/mleap/core/feature/MaxAbsScalerModelSpec.scala:16]: 0.9500000000000001 make test flaky ? Why not use use eq with error tolerance ?

Comment (mengxr) [reply mleap-core/src/test/scala/ml/combust/mleap/core/feature/MaxAbsScalerModelSpec.scala:16]: We need to refactor the parity check to allow tolerance. It is comparing row by row right now.

File: mleap-core/src/main/scala/ml/combust/mleap/core/feature/MinMaxScalerModel.scala [hunk 2]
```diff
@@ -15,38 +15,49 @@ import scala.math.{max, min}
   * @param originalMin minimum values from training features
   * @param originalMax maximum values from training features
   */
-@SparkCode(uri = "https://github.com/apache/spark/blob/v2.4.0/mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala")
+@SparkCode(uri = "https://github.com/apache/spark/blob/v3.0.0-rc1/mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala")
```
[the remark this answers is not in the document]
Comment (mengxr) [reply mleap-core/src/main/scala/ml/combust/mleap/core/feature/MinMaxScalerModel.scala:18]: Just a reference to track the original of the code

File: project/Dependencies.scala [hunk 3]
```diff
@@ -6,8 +6,8 @@ import Keys._
 object Dependencies {
   import DependencyHelpers._
 
-  val sparkVersion = "2.4.0"
-  val scalaTestVersion = "3.0.0"
+  val sparkVersion = "3.0.0-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (mengxr) [reply project/Dependencies.scala:9]: This PR is not ready for merge

**Comment by ancasarb** [comment]
hey @mengxr I'll take a look at the changes this week, just to let you know, there's a couple of PRs that I would like to get merged before (i am currently reviewing them) and do a release with spark 2.4 before getting the upgrade to spark 3, when it's ready. i hope this is ok.

**Comment by mengxr** [comment]
@ancasarb That is okay. This PR is not ready for merge, not until we have cross builds and backward compatibility working.

**Comment by mengxr** [comment]
Closing this in favor of #665 .

**Pull request closed without merging**